src/Core/Domain/State/Command/ToggleStateStatusCommand.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\State\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\StateConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\State\ValueObject\StateId;
+
+/**
+ * Toggles state status in row action
+ */
+class ToggleStateStatusCommand
+{
+    /**
+     * @var StateId
+     */
+    private $stateId;
+
+    /**
+     * @var bool
+     */
+    private $expectedStatus;
+
+    /**
+     * @param int $stateId
+     * @param bool $expectedStatus
+     *
+     * @throws StateConstraintException
+     */
+    public function __construct(int $stateId, bool $expectedStatus)
```

Comment:
See https://github.com/PrestaShop/PrestaShop/issues/17407
But I think this is the right answer

Reply:
As I understood the question is do we really need to pass $expectedStatus? It is not needed if we use objectModel->toggleStatus();